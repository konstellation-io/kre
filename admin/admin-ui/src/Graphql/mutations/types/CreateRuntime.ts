/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import {
  CreateRuntimeInput,
  RuntimeStatus,
  VersionStatus
} from '../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: CreateRuntime
// ====================================================

export interface CreateRuntime_createRuntime_creationAuthor {
  __typename: 'User';
  id: string;
  email: string;
}

export interface CreateRuntime_createRuntime_publishedVersion {
  __typename: 'Version';
  status: VersionStatus;
}

export interface CreateRuntime_createRuntime {
  __typename: 'Runtime';
  id: string;
  name: string;
  description: string;
  status: RuntimeStatus;
  creationDate: string;
  creationAuthor: CreateRuntime_createRuntime_creationAuthor;
  publishedVersion: CreateRuntime_createRuntime_publishedVersion | null;
}

export interface CreateRuntime {
  createRuntime: CreateRuntime_createRuntime;
}

export interface CreateRuntimeVariables {
  input: CreateRuntimeInput;
}
