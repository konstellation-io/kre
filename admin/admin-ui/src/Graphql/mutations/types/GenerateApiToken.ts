/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: GenerateApiToken
// ====================================================

export interface GenerateApiToken {
  generateApiToken: string;
}
