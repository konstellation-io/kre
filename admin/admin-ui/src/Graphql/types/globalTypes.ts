/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum AccessLevel {
  ADMIN = 'ADMIN',
  MANAGER = 'MANAGER',
  VIEWER = 'VIEWER'
}

export enum ConfigurationVariableType {
  FILE = 'FILE',
  VARIABLE = 'VARIABLE'
}

export enum LogLevel {
  DEBUG = 'DEBUG',
  ERROR = 'ERROR',
  INFO = 'INFO',
  WARN = 'WARN'
}

export enum NodeStatus {
  ERROR = 'ERROR',
  STARTED = 'STARTED',
  STOPPED = 'STOPPED'
}

export enum RuntimeStatus {
  CREATING = 'CREATING',
  ERROR = 'ERROR',
  STARTED = 'STARTED'
}

export enum UserActivityType {
  CREATE_RUNTIME = 'CREATE_RUNTIME',
  CREATE_USER = 'CREATE_USER',
  CREATE_VERSION = 'CREATE_VERSION',
  LOGIN = 'LOGIN',
  LOGOUT = 'LOGOUT',
  PUBLISH_VERSION = 'PUBLISH_VERSION',
  REMOVE_USERS = 'REMOVE_USERS',
  REVOKE_SESSIONS = 'REVOKE_SESSIONS',
  START_VERSION = 'START_VERSION',
  STOP_VERSION = 'STOP_VERSION',
  UNPUBLISH_VERSION = 'UNPUBLISH_VERSION',
  UPDATE_ACCESS_LEVELS = 'UPDATE_ACCESS_LEVELS',
  UPDATE_SETTING = 'UPDATE_SETTING',
  UPDATE_VERSION_CONFIGURATION = 'UPDATE_VERSION_CONFIGURATION'
}

export enum VersionStatus {
  PUBLISHED = 'PUBLISHED',
  STARTED = 'STARTED',
  STARTING = 'STARTING',
  STOPPED = 'STOPPED'
}

export interface ConfigurationVariablesInput {
  key: string;
  value: string;
}

export interface CreateRuntimeInput {
  name: string;
  description: string;
}

export interface CreateUserInput {
  email: string;
  accessLevel: AccessLevel;
}

export interface CreateVersionInput {
  file: any;
  runtimeId: string;
}

export interface LogFilters {
  startDate: string;
  endDate?: string | null;
  search?: string | null;
  levels?: LogLevel[] | null;
  nodeIds?: string[] | null;
}

export interface PublishVersionInput {
  versionId: string;
  comment: string;
}

export interface SettingsInput {
  authAllowedDomains?: string[] | null;
  sessionLifetimeInDays?: number | null;
}

export interface StartVersionInput {
  versionId: string;
  comment: string;
}

export interface StopVersionInput {
  versionId: string;
  comment: string;
}

export interface UnpublishVersionInput {
  versionId: string;
  comment: string;
}

export interface UpdateAccessLevelInput {
  userIds: string[];
  accessLevel: AccessLevel;
  comment: string;
}

export interface UpdateConfigurationInput {
  versionId: string;
  configurationVariables: ConfigurationVariablesInput[];
}

export interface UsersInput {
  userIds: string[];
  comment: string;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
