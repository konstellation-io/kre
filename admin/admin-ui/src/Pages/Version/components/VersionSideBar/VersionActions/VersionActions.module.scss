@import 'Styles/colors';
@import 'Styles/mixins';

.wrapper {
  color: font-color(light);
  display: flex;
  flex-direction: row;
  background-color: #26292c;
  align-items: center;
  margin-top: auto;

  &.quickActions {
    background-color: transparent;
  }

  .item {
    @include font-body;
    font-weight: 600;

    height: 48px;
    padding-left: 16px;
    display: flex;
    flex-direction: row;
    align-items: center;
    flex-grow: 1;
    cursor: pointer;

    .icon {
      margin-right: 8px;
      line-height: 0;
    }

    &:hover {
      color: font-color(light);
    }
  }
}

.disabled {
  cursor: default;
  pointer-events: none;
  color: font-color(regular);
}

.warning {
  @include font-small;
  color: palette(alert);

  margin-bottom: $grid-unit;

  .publishedVersion {
    text-transform: uppercase;
    color: font-color(light);
  }
}
