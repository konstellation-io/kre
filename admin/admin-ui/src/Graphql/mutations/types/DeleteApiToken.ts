/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { DeleteApiTokenInput } from './../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: DeleteApiToken
// ====================================================

export interface DeleteApiToken_deleteApiToken {
  __typename: 'ApiToken';
  id: string;
}

export interface DeleteApiToken {
  deleteApiToken: DeleteApiToken_deleteApiToken;
}

export interface DeleteApiTokenVariables {
  input: DeleteApiTokenInput;
}
