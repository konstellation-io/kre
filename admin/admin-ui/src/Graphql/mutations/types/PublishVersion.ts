/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import {PublishVersionInput, VersionStatus} from '../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: PublishVersion
// ====================================================

export interface PublishVersion_publishVersion_publicationAuthor {
  __typename: 'User';
  email: string;
  id: string;
}

export interface PublishVersion_publishVersion {
  __typename: 'Version';
  id: string;
  status: VersionStatus;
  publicationAuthor: PublishVersion_publishVersion_publicationAuthor | null;
}

export interface PublishVersion {
  publishVersion: PublishVersion_publishVersion;
}

export interface PublishVersionVariables {
  input: PublishVersionInput;
}
