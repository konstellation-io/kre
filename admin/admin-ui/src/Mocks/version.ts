import { RuntimeStatus, VersionStatus } from 'Graphql/types/globalTypes';

import { loader } from 'graphql.macro';

const GetConfigurationVariablesQuery = loader(
  'Graphql/queries/getConfigurationVariables.graphql'
);
const GetVersionWorkflowsQuery = loader(
  'Graphql/queries/getVersionWorkflows.graphql'
);
const GetVersionNodeStatusQuery = loader(
  'Graphql/subscriptions/versionNodeStatus.graphql'
);
const NodeStatusSubscription = loader(
  'Graphql/subscriptions/versionNodeStatus.graphql'
);
const PublishVersionMutation = loader(
  'Graphql/mutations/publishVersion.graphql'
);
const StartVersionMutation = loader('Graphql/mutations/startVersion.graphql');
const StopVersionMutation = loader('Graphql/mutations/stopVersion.graphql');
const UnpublishVersionMutation = loader(
  'Graphql/mutations/unpublishVersion.graphql'
);

export const runtime = {
  id: 'runtimeId',
  name: 'runtimeName',
  description: 'Some Description',
  creationAuthor: { email: 'some@user.com' },
  status: RuntimeStatus.STARTED,
  __typename: 'Runtime'
};
export const version = {
  __typename: 'Version',
  id: 'versionId',
  name: 'versionName',
  description: 'versionDescription',
  status: VersionStatus.STARTED,
  creationDate: '2020/01/01',
  creationAuthor: {
    __typename: 'User',
    id: 'userId',
    email: 'user@domain.com'
  },
  publicationDate: '2020/01/01',
  publicationAuthor: {
    __typename: 'User',
    id: 'userId',
    email: 'user@domain.com'
  },
  config: {
    vars: [],
    completed: false
  }
};
export const confVarsMock = {
  request: {
    query: GetConfigurationVariablesQuery
  },
  result: {
    data: {
      version: {
        status: 'STARTED',
        config: {
          completed: true,
          vars: [
            {
              key: 'var2',
              value: 'value1',
              type: 'VARIABLE'
            },
            {
              key: 'var1',
              value: 'value2',
              type: 'VARIABLE'
            },
            {
              key: 'var3',
              value: 'value3',
              type: 'FILE'
            }
          ]
        }
      }
    }
  }
};

export const workflowsMock = {
  request: {
    query: GetVersionWorkflowsQuery
  },
  result: {
    data: {
      version: {
        name: 'version01',
        status: 'STARTED',
        configurationCompleted: true,
        workflows: [
          {
            name: 'workflow01',
            nodes: [
              {
                id: 'node01',
                name: 'one node',
                status: 'STARTED'
              },
              {
                id: 'node02',
                name: 'another node',
                status: 'STARTED'
              }
            ],
            edges: [
              {
                id: 'edge01',
                fromNode: 'node01',
                toNode: 'node02'
              }
            ]
          }
        ]
      }
    }
  }
};

export const errorMorkflowsMock = {
  request: {
    query: GetVersionWorkflowsQuery
  },
  error: 'Some error'
};

export const nodeStatus = {
  request: {
    query: GetVersionNodeStatusQuery
  },
  result: {
    data: {
      versionNodeStatus: [
        {
          date: '2020-01-01',
          nodeId: 'nodeId1',
          status: 'STARTED',
          message: 'some message 1'
        },
        {
          date: '2020-01-01',
          nodeId: 'nodeId2',
          status: 'STARTED',
          message: 'some message 2'
        }
      ]
    }
  }
};

export const nodeStatusMock = {
  request: {
    query: NodeStatusSubscription
  },
  result: {
    data: {
      versionNodeStatus: {
        date: '2020/01/01',
        nodeId: 'node02',
        status: 'STARTED',
        message: 'some message'
      }
    }
  }
};

export const publishVersionMock = {
  request: {
    query: PublishVersionMutation
  },
  result: {
    data: {
      version
    }
  }
};

export const startVersionMock = {
  request: {
    query: StartVersionMutation
  },
  result: {
    data: {
      version
    }
  }
};

export const stopVersionMock = {
  request: {
    query: StopVersionMutation
  },
  result: {
    data: {
      version
    }
  }
};

export const unpublishVersionMock = {
  request: {
    query: UnpublishVersionMutation
  },
  result: {
    data: {
      version
    }
  }
};
