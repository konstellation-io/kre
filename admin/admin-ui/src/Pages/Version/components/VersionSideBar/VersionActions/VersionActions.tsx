import {
  BUTTON_THEMES,
  Button,
  CHECK,
  ModalContainer,
  ModalLayoutInfo,
  ModalLayoutJustify
} from 'kwc';
import {
  GetVersionConfStatus_runtime,
  GetVersionConfStatus_versions
} from 'Graphql/queries/types/GetVersionConfStatus';
import React, { FunctionComponent, useEffect, useRef, useState } from 'react';
import Tag, { TagTypes } from 'Components/Tag/Tag';

import { Link } from 'react-router-dom';
import PublishIcon from '@material-ui/icons/Publish';
import ROUTE from 'Constants/routes';
import StartIcon from '@material-ui/icons/PlayCircleOutline';
import StopIcon from '@material-ui/icons/PauseCircleFilled';
import { SvgIconProps } from '@material-ui/core/SvgIcon';
import UnpublishIcon from '@material-ui/icons/GetApp';
import { VersionStatus } from 'Graphql/types/globalTypes';
import { buildRoute } from 'Utils/routes';
import cx from 'classnames';
import { get } from 'lodash';
import styles from './VersionActions.module.scss';
import { useForm } from 'react-hook-form';
import useVersionAction from '../../../utils/hooks';

function verifyComment(value: string) {
  return CHECK.getValidationError([CHECK.isFieldNotEmpty(value)]);
}

type WarningProps = {
  runtime: GetVersionConfStatus_runtime;
  publishedVersion: GetVersionConfStatus_versions;
};
function Warning({ runtime, publishedVersion }: WarningProps) {
  return (
    <>
      <Tag type={TagTypes.WARNING}>WARNING</Tag>A published version already
      exists, publishing this version will unpublish the following version:{' '}
      <Link
        to={buildRoute.version(
          ROUTE.RUNTIME_VERSION,
          runtime.id,
          publishedVersion.id
        )}
      >
        <span
          className={styles.publishedVersion}
          title={publishedVersion.description}
        >
          {publishedVersion.name}
        </span>
      </Link>
    </>
  );
}

type ModalInfo = {
  action: (comment: string) => void;
  label: string;
};

type ActionProps = {
  Icon: FunctionComponent<SvgIconProps>;
  label: string;
  action?: Function;
  disabled?: boolean;
  primary?: boolean;
};

type FormData = {
  comment: string;
};

type Props = {
  runtime: GetVersionConfStatus_runtime;
  versions: GetVersionConfStatus_versions[];
  version: GetVersionConfStatus_versions;
  quickActions?: boolean;
};

function VersionActions({
  runtime,
  versions,
  version,
  quickActions = false
}: Props) {
  const { handleSubmit, setValue, register, unregister, errors } = useForm<
    FormData
  >({
    defaultValues: {
      comment: ''
    }
  });
  const [showConfirmation, setShowConfirmation] = useState(false);
  const modalInfo = useRef<ModalInfo>({
    action: () => {},
    label: ''
  });

  useEffect(() => {
    register('comment', { validate: verifyComment });

    return () => unregister('comment');
  }, [register, unregister, setValue]);

  function onSubmit() {
    handleSubmit((formData: FormData) =>
      modalInfo.current.action(formData.comment)
    )();
  }

  const {
    publishVersion,
    startVersion,
    stopVersion,
    unpublishVersion,
    getMutationVars
  } = useVersionAction(runtime.id);

  function openModal(label: string, mutation: Function) {
    modalInfo.current = {
      action: (comment: string) => {
        mutation(getMutationVars(version.id, comment));
        closeModal();
      },
      label
    };
    setShowConfirmation(true);
  }

  function closeModal() {
    setShowConfirmation(false);
  }

  let buttons: ActionProps[] = [];

  switch (version.status) {
    case VersionStatus.CREATED:
    case VersionStatus.STOPPED:
      buttons[0] = {
        Icon: StartIcon,
        label: 'START',
        action: () => openModal('START', startVersion),
        primary: version.config.completed,
        disabled: !version.config.completed
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        disabled: true
      };
      break;
    case VersionStatus.STOPPING:
      buttons[0] = {
        Icon: StartIcon,
        label: 'START',
        action: () => openModal('START', startVersion),
        primary: version.config.completed,
        disabled: true
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        disabled: true
      };
      break;
    case VersionStatus.CREATING:
      buttons[0] = {
        Icon: StartIcon,
        label: 'START',
        disabled: true
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        disabled: true
      };
      break;
    case VersionStatus.STARTING:
      buttons[0] = {
        Icon: StopIcon,
        label: 'STOP',
        action: () => openModal('STOP', stopVersion),
        disabled: true
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        disabled: true
      };
      break;
    case VersionStatus.STARTED:
      buttons[0] = {
        Icon: StopIcon,
        label: 'STOP',
        action: () => openModal('STOP', stopVersion)
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        action: () => openModal('PUBLISH', publishVersion),
        primary: true
      };
      break;
    case VersionStatus.PUBLISHED:
      buttons[0] = {
        Icon: StopIcon,
        label: 'STOP',
        disabled: true
      };
      buttons[1] = {
        Icon: UnpublishIcon,
        label: 'UNPUBLISH',
        action: () => openModal('UNPUBLISH', unpublishVersion),
        primary: true
      };
      break;
  }

  if (quickActions) {
    buttons = buttons.filter(button => !button.disabled);
  }

  const publishedVersion = versions.filter(
    v => v.status === VersionStatus.PUBLISHED
  )[0];

  const showWarning = publishedVersion && modalInfo.current.label === 'PUBLISH';

  return (
    <div
      className={cx(styles.wrapper, { [styles.quickActions]: quickActions })}
    >
      {buttons.map(btn => (
        <Button
          key={btn.label}
          onClick={() => btn.action && btn.action()}
          label={quickActions ? '' : btn.label}
          title={quickActions ? btn.label : ''}
          Icon={btn.Icon}
          iconSize="icon-regular"
          theme={
            quickActions ? BUTTON_THEMES.TRANSPARENT : BUTTON_THEMES.DEFAULT
          }
          disabled={btn.disabled}
          primary={quickActions ? false : btn.primary}
          style={{ flexGrow: 1 }}
        />
      ))}
      {showConfirmation && (
        <ModalContainer
          title={`YOU ARE ABOUT TO ${modalInfo.current.label} A VERSION`}
          actionButtonLabel={modalInfo.current.label}
          onAccept={onSubmit}
          onCancel={closeModal}
          warning={showWarning}
          blocking
        >
          <ModalLayoutJustify
            onUpdate={(value: string) => setValue('comment', value)}
            submit={onSubmit}
            error={get(errors.comment, 'message', '') as string}
          />
          {showWarning && (
            <ModalLayoutInfo className={styles.warning}>
              <Warning runtime={runtime} publishedVersion={publishedVersion} />
            </ModalLayoutInfo>
          )}
        </ModalContainer>
      )}
    </div>
  );
}

export default VersionActions;
