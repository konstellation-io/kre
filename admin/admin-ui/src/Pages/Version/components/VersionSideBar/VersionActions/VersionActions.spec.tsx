import { Button, ModalContainer, ModalLayoutJustify } from 'kwc';
import { runtime, version } from 'Mocks/version';

import React from 'react';
import VersionActions from './VersionActions';
import { VersionStatus } from 'Graphql/types/globalTypes';
import { clone } from 'lodash';
import { label } from 'Utils/testUtilsEnzyme';
import { shallow } from 'enzyme';

jest.mock('@apollo/client', () => ({
  useMutation: jest.fn(() => [jest.fn(), { loading: false }])
}));

describe('VersionActions', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <VersionActions runtime={runtime} version={version} versions={[]} />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has right components', async () => {
    expect(wrapper.exists('.wrapper')).toBeTruthy();
    expect(wrapper.find(Button).length).toBe(2);
  });

  it('show right buttons when version is STOPPED', async () => {
    const versionStopped = clone(version);
    versionStopped.status = VersionStatus.STOPPED;

    wrapper.setProps({ version: versionStopped });

    expect(wrapper.find(Button).length).toBe(2);
    expect(wrapper.find(Button).exists({ label: 'START' })).toBeTruthy();
    expect(wrapper.find(Button).exists({ label: 'PUBLISH' })).toBeTruthy();
    expect(
      wrapper
        .find(Button)
        .find({ label: 'PUBLISH' })
        .prop('disabled')
    ).toBeTruthy();
  });

  it('show START button disabled when version is STOPPED and not configured', async () => {
    const versionStopped = clone(version);
    versionStopped.status = VersionStatus.STOPPED;
    versionStopped.configurationCompleted = false;

    wrapper.setProps({ version: versionStopped });

    expect(wrapper.find(Button).exists({ label: 'START' })).toBeTruthy();
    expect(
      wrapper
        .find(Button)
        .find({ label: 'START' })
        .prop('disabled')
    ).toBeTruthy();
  });

  it('show right buttons when version is PUBLISHED', async () => {
    const versionPublished = clone(version);
    versionPublished.status = VersionStatus.PUBLISHED;

    wrapper.setProps({ version: versionPublished });

    expect(wrapper.find(Button).length).toBe(2);
    expect(wrapper.find(Button).exists({ label: 'STOP' })).toBeTruthy();
    expect(
      wrapper
        .find(Button)
        .find({ label: 'STOP' })
        .prop('disabled')
    ).toBeTruthy();
    expect(wrapper.find(Button).exists({ label: 'UNPUBLISH' })).toBeTruthy();
  });

  it('show right buttons when version is STARTING', async () => {
    const versionStarting = clone(version);
    versionStarting.status = VersionStatus.STARTING;

    wrapper.setProps({ version: versionStarting });

    expect(wrapper.find(Button).length).toBe(2);
  });

  it('show right buttons when version is STARTED', async () => {
    expect(wrapper.find(Button).length).toBe(2);
    expect(wrapper.find(Button).exists({ label: 'STOP' })).toBeTruthy();
    expect(wrapper.find(Button).exists({ label: 'PUBLISH' })).toBeTruthy();
  });

  it('shows confirmation modal when clicking PUBLISH button', async () => {
    wrapper
      .find(Button)
      .find(label('PUBLISH'))
      .simulate('click');

    expect(wrapper.exists(ModalLayoutJustify)).toBeTruthy();
  });

  it('hides confirmation modal on close', async () => {
    wrapper
      .find(Button)
      .find(label('PUBLISH'))
      .simulate('click');

    wrapper
      .find(ModalContainer)
      .dive()
      .find(label('CANCEL'))
      .simulate('click');

    expect(wrapper.exists(ModalContainer)).toBeFalsy();
  });
});
