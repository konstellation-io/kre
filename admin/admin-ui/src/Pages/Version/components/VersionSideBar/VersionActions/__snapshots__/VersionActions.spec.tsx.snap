// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionActions matches snapshot 1`] = `
<div
  className="wrapper"
>
  <Button
    Icon={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "displayName": "PauseCircleFilledIcon",
        "muiName": "SvgIcon",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      }
    }
    iconSize="icon-regular"
    key="STOP"
    label="STOP"
    onClick={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    theme="default"
    title=""
  />
  <Button
    Icon={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "displayName": "PublishIcon",
        "muiName": "SvgIcon",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      }
    }
    iconSize="icon-regular"
    key="PUBLISH"
    label="PUBLISH"
    onClick={[Function]}
    primary={true}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    theme="default"
    title=""
  />
</div>
`;
